core/trino-main/src/test/java/io/trino/operator/TestExchangeOperator.java

```diff
@@ -149,7 +152,7 @@ public void testSimple()
 
     private static Split newRemoteSplit(TaskId taskId)
     {
-        return new Split(REMOTE_CONNECTOR_ID, new RemoteSplit(taskId, URI.create("http://localhost/" + taskId)), Lifespan.taskWide());
+        return new Split(REMOTE_CONNECTOR_ID, new RemoteSplit(new DirectExchangeInput(taskId, URI.create("http://localhost/" + taskId))), Lifespan.taskWide());
```

Comment:
The tests in the `TestExchangeOperator` rely heavily on the streaming mode of execution (http requests are emulated) thus it might be difficult to extract some common bits. I was also thinking about implementing a specific test, but it would likely require a lot of mocking to the point where it is unclear if there's a real value. Though I don't have a strong opinion here.

Reply:
Let's skip it. Integration tests should be enough. There is of course possibility that we are missing some coverage but if we spot that we should still try to cover it with integration tests if possible. Let's only add testing at unit level here as a last resort.